CMakeLists.txt

```diff
@@ -33,11 +33,28 @@ if (${VARA_FEATURE_COLORED_OUTPUT})
   endif ()
 endif ()
 
+option(VARA_FEATURE_SANITIZE "Build vara-features with ASan and UBSan" FALSE)
+if (${VARA_FEATURE_SANITIZE})
+  set(CMAKE_CXX_FLAGS "-fsanitize=address,undefined ${CMAKE_CXX_FLAGS}")
+else ()
+  set(CMAKE_CXX_FLAGS "-fno-sanitize=address,undefined ${CMAKE_CXX_FLAGS}")
+endif ()
+
 option(VARA_FEATURE_BUILD_PYTHON_BINDINGS "Build vara-features python bindings" TRUE)
 if (${VARA_FEATURE_BUILD_PYTHON_BINDINGS})
+  if (${VARA_FEATURE_SANITIZE})
+    message(FATAL_ERROR "Cannot build python bindings together with sanitizers.")
+  endif ()
   add_compile_options(-fPIC)
 endif ()
 
+# Extra debug warning flags
+if (CMAKE_BUILD_TYPE EQUAL "Debug")
+  add_compile_options(-Wall -Wextra -Wno-unused-parameter -O0)
+elseif (CMAKE_BUILD_TYPE EQUAL "Release")
+  add_compile_options(-O3)
```

Comment:
```suggestion
```
I'm not sure but is this necessary? As far as I know `Release` should automatically add -O3, maybe I'm wrong.

Reply:
You are right, default values are
```
CMAKE_CXX_FLAGS_DEBUG='-g'
CMAKE_CXX_FLAGS_RELEASE='-O3 -DNDEBUG'
```